coalton-asdf.asd

```diff
@@ -0,0 +1,23 @@
+;;; Coalton ASDF extensions
```

Comment:
Is there a reason this needs to be a separate asdf system?

Reply:
Yes, there is a cyclic load error, because secondary systems are presumed dependencies of the primary system, which prevents the system from accepting coalton/compiler as a dependency.

```
Unhandled LOAD-SYSTEM-DEFINITION-ERROR in thread #<SB-THREAD:THREAD tid=259 "main thread" RUNNING {7005000003}>: Error while trying to load definition for system coalton from pathname /Users/jlbouwman/git/coalton/coalton.asd:
Circular dependency:
((#<DEFINE-OP > . #<SYSTEM "coalton">)
 (#<LOAD-OP > . #<SYSTEM "coalton/asdf">)
 (#<LOAD-OP > . #<CL-SOURCE-FILE "coalton/asdf" "coalton-asdf">)
 (#<PREPARE-OP > . #<CL-SOURCE-FILE "coalton/asdf" "coalton-asdf">)
 (#<PREPARE-OP > . #<SYSTEM "coalton/asdf">)
 (#<LOAD-OP > . #<SYSTEM "coalton/compiler">)
 (#<LOAD-OP > . #<CL-SOURCE-FILE "coalton/compiler" "package">)
 (#<PREPARE-OP > . #<CL-SOURCE-FILE "coalton/compiler" "package">)
 (#<PREPARE-OP > . #<SYSTEM "coalton/compiler">))
```